packages/components/src/components/drawer/drawer.jsx

```diff
@@ -0,0 +1,87 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class Drawer extends React.PureComponent {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            is_open: true,
+        };
+    }
+
+    toggleDrawer = () => {
+        this.setState({ is_open: !this.state.is_open });
```

Comment:
need to use previous state here:

```suggestion
this.setState(prev_state => ({
    is_open: !prev_state.is_open
}));
```